loony_dev/tests/test_observe.py

```
@@ -213,6 +213,67 @@ def test_observe_jsonl_path_none_for_unknown(self) -> None:
+        self._write_conn({"cwd": "relative/path", "session_id": "sid"})
+        self.assertIsNone(services.live_observe_jsonl_path(self.base, "acme", "widgets"))
+
+    def test_none_for_traversal_segment(self) -> None:
+        self.assertIsNone(services.live_observe_jsonl_path(self.base, "..", "widgets"))
+        self.assertIsNone(services.live_observe_jsonl_path(self.base, "acme", "a/b"))
+
+    def test_none_for_path_bearing_session_id(self) -> None:
+        # session_id becomes the transcript filename in jsonl_path_for(), so a
+        # path-bearing value from a malformed connection file must be rejected
+        # (→ honest None) rather than escaping projects/<slug>/.
+        for sid in ("..", ".", "../other", "/tmp/x", "a/b", "a\\b", "a\x00b"):
```

Ruff (0.15.20)

[error] Probable insecure usage of temporary file or directory: "/tmp/x"

(S108)